Test fails w/ latest CTAN snapshot (babel/greek)

I checked out TeX Live another time at end of June and noticed that LaTeXML again fails to run a test using that TL snapshot.

```
t/81_babel.t ..............
1..7
ok 1 - use LaTeXML::Core;
ok 2 - t/babel/csquotes
ok 3 - t/babel/french
ok 4 - t/babel/german
not ok 5 - t/babel/greek

#   Failed test 't/babel/greek'
#   at /home/hille/latexml-0.8.8-pre/blib/lib/LaTeXML/Util/Test.pm line 120.
Wide character in print at /usr/share/perl/5.36/Test2/Formatter/TAP.pm line 125.
# Difference at line 41 for t/babel/greek
#       got : '<p>Ηερε´ς'
#  expected : '<p>Here’s'

Test Summary Report
-------------------
t/81_babel.t            (Wstat: 256 (exited 1) Tests: 7 Failed: 1)
  Failed test:  5
  Non-zero exit status: 1
Files=34, Tests=443, 524 wallclock secs ( 0.24 usr  0.12 sys + 492.41 cusr 29.91 csys = 522.68 CPU)
Result: FAIL
Failed 1/34 test programs. 1/443 subtests failed.
make: *** [Makefile:1848: test_dynamic] Error 255
```
This is reproducible with a git clone of LaTeXML I made this morning. The full build log can be seen [here](https://ci.debian.net/data/autopkgtest/unstable/amd64/l/latexml/36138938/log.gz).

Sorry for the bad news! Thanks for help.

--- a/lib/LaTeXML/Core/Mouth.pm
+++ b/lib/LaTeXML/Core/Mouth.pm
@@ -20,7 +20,7 @@ use LaTeXML::Core::Token;
 use LaTeXML::Core::Tokens;
 use LaTeXML::Util::Pathname;
 use Encode qw(decode);
-use base   qw(LaTeXML::Common::Object);
+use base qw(LaTeXML::Common::Object);
 
 our $READLINE_PROGRESS_QUANTUM = 25;
 
@@ -214,7 +214,7 @@ sub handle_escape {    # Read control sequence
   # Bit I believe that he does NOT mean within control sequences
   my $cs = "\\" . $ch;    # I need this standardized to be able to lookup tokens (A better way???)
   if ((defined $cc) && ($cc == CC_LETTER)) {    # For letter, read more letters for csname.
-    while ((($ch, $cc) = getNextChar($self)) && $ch && ($cc == CC_LETTER)) {
+    while ((($ch, $cc) = getNextChar($self)) && (length($ch) > 0) && ($cc == CC_LETTER)) {
       $cs .= $ch; }
     # We WILL skip spaces, but not till next token is read (in case catcode changes!!!!)
     $$self{skipping_spaces} = 1;
@@ -321,10 +321,10 @@ sub readToken {
       my ($ch, $cc);
       while ((($ch, $cc) = getNextChar($self)) && (defined $ch)
         && (($cc == CC_SPACE) || ($cc == CC_IGNORE))) { }
-      if ($ch && ($cc == CC_EOL)) {    # Eolch already? empty line!
-        $$self{colno} = $$self{nchars};    # ignore rest of line.
+      if ((defined $ch) && ($cc == CC_EOL)) {    # Eolch already? empty line!
+        $$self{colno} = $$self{nchars};          # ignore rest of line.
         return T_CS('\par'); }
-      elsif ($$self{colno} > $$self{nchars}) {    # Past end of line?
+      elsif (($$self{nchars} == 0) || ($$self{colno} > $$self{nchars})) {    # Past end of line?
             # If upcoming line is empty, and there is no recognizable EOL, fake one
         return T_MARKER('EOL') if $read_mode && ((!defined $eolch) || ($eolch ne "\r")); }
       else {    # Back up over peeked char
@@ -347,7 +347,6 @@ sub readToken {
     my $token = (defined $cc ? $DISPATCH[$cc] : undef);
     $token = &$token($self, $ch) if ref $token eq 'CODE';
     return $token if defined $token;    # Else, repeat till we get something or run out.
-
   }
   return; }
 

--- a/lib/LaTeXML/Core/Mouth.pm
+++ b/lib/LaTeXML/Core/Mouth.pm
@@ -214,7 +214,7 @@ sub handle_escape {    # Read control sequence
   # Bit I believe that he does NOT mean within control sequences
   my $cs = "\\" . $ch;    # I need this standardized to be able to lookup tokens (A better way???)
   if ((defined $cc) && ($cc == CC_LETTER)) {    # For letter, read more letters for csname.
-    while ((($ch, $cc) = getNextChar($self)) && (length($ch) > 0) && ($cc == CC_LETTER)) {
+    while ((($ch, $cc) = getNextChar($self)) && (defined $ch) && ($cc == CC_LETTER)) {
       $cs .= $ch; }
     # We WILL skip spaces, but not till next token is read (in case catcode changes!!!!)
     $$self{skipping_spaces} = 1;

